src/components/WalletDropdown/TransactionMenu.tsx

```diff
@@ -1,9 +1,159 @@
 import { Trans } from '@lingui/macro'
+import { useWeb3React } from '@web3-react/core'
+import { getYear, isSameDay, isSameWeek, isSameYear } from 'date-fns'
+import { useCallback, useMemo } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import styled from 'styled-components/macro'
 
+import { useAllTransactions } from '../../state/transactions/hooks'
+import { clearAllTransactions } from '../../state/transactions/reducer'
+import { TransactionDetails } from '../../state/transactions/types'
+import { TransactionSummary } from '../AccountDetailsV2'
 import { SlideOutMenu } from './SlideOutMenu'
 
-export const TransactionHistoryMenu = ({ onClose }: { onClose: () => void }) => (
-  <SlideOutMenu onClose={onClose} onClear={undefined} title={<Trans>Transactions</Trans>}>
-    <div />
-  </SlideOutMenu>
-)
+const Divider = styled.div`
+  margin-top: 16px;
+  border-bottom: ${({ theme }) => `1px solid ${theme.backgroundOutline}`};
+`
+
+const TransactionListWrapper = styled.div`
+  ${({ theme }) => theme.flexColumnNoWrap};
+`
+
+interface TransactionInformation {
+  title: string
+  transactions: TransactionDetails[]
+}
+
+const TransactionTitle = styled.span`
+  padding-bottom: 8px;
+  padding-top: 20px;
+  padding-left: 12px;
+  padding-right: 12px;
+  font-weight: 600;
+  color: ${({ theme }) => theme.textTertiary};
+`
+
+const renderTransactions = (transactionInformation: TransactionInformation) => {
```

Comment:
Would make this a component called `TransactionList`. generally don't want a function that returns JSX as you lose the React perf benefits